packages/common/src/reducers/apps.js

```diff
@@ -0,0 +1,20 @@
+import { createSelector } from "reselect";
+import { indexBy } from "../utils/array";
+import combineReducers from "../utils/combine-reducers";
+import { arrayShallowEquals } from "../utils/reselect";
+
+const entriesById = (state = {}, action) => {
+  switch (action.type) {
+    case "initial-data-request-successful": {
+      if (!action.data.apps) return state;
+      return indexBy((a) => a.id, action.data.apps);
+    }
+
+    default:
+      return state;
+  }
+};
+
+export const selectApp = (state, appId) => state.apps?.entriesById[appId];
```

Comment:
Shouldn’t need the question mark here, or we have a bug

Reply:
Then the `entriesById` is just an empty object. The full reducer is run once on startup with dummy action, so the initial state will be the default argument (`state = {}` in this case)